client/state/data-layer/wpcom/concierge/schedules/shifts/from-api.js

```diff
@@ -0,0 +1,18 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { makeParser } from 'state/data-layer/wpcom-http/utils';
+import responseSchema from './schema';
+
+const transformShift = shift => ( {
+	beginTimestamp: shift.begin_timestamp,
+	endTimestamp: shift.end_timestamp,
+	scheduleId: shift.schedule_id,
+	description: shift.description,
+} );
```

Comment:
Did you consider using a utility like [camelCaseKeysDeep](https://github.com/abhishekgahlot/gutenberg/blob/84a16809f73f9c065051b6fc9748becf4059a0d9/utils/lang.js#L6) when the only change you want to introduce in your data is to change the keys?

Reply:
without trying to contradict the suggestion, I'll note that having the data that we use explicitly laid out here is one incredibly helpful way of communicating to other developers what information is available on the data introduced into Calypso - it's a whitelist and if we wanted to use another property from the API response we'd have to come and add it here then everyone will see that it's changing.

Reply:
A better solution is typescript :).  If a new property gets added you are forced to add it to the type and it doesn't end up bloating the code

Reply:
@gziolo I personally prefer to only pick the expected fields here. I've just added a test case to demonstrate the attempt clearer in 87f7834 :)

Does it make sense to you?

Reply:
I hope the following makes my intention clearer:

```js
camelCaseKeysDeep(
    pick(
        shift,
       [ 'begin_timestamp', 'end_timestamp', 'schedule_id', 'description' ]
    )
);

// or

pick(
    camelCaseKeysDeep( shift ),
   [ 'beginTimestamp', 'endTimestamp', 'scheduleId', 'description' ]
);
```

The more fields you want to pick, the more keystrokes you save. It's your call. I'm only saying that there are alternative ways of tackling this.